Create a badge for the daily test runs in travis-ci

JamesWTruher opened the pull request:
It updates an azure blob with an SVG and that location is referenced by the
README.md file. TravisCI doesn't support this directly so in order to report
on the status of a daily test run there, we need to do this.
Provide error reporting in case the badge can't be set

The location where the badge is stored in azure should be moved eventually to a global resource where we keep other azure blobs. We don't have that yet, but this will do. Also, the location in azure is globally readable, but not globally writable. Travis-CI configuration must be updated to include the two required environment variables.

JamesWTruher committed:
changes which enable setting a badge for the daily test runs in travis

It updates an azure blob with an SVG and that location is referenced by the
README.md file. TravisCI doesn't support this directly so in order to report
on the status of a daily test run there, we need to do this.
Provide error reporting in case the badge can't be set

adityapatwardhan commented on tools/travis.ps1:
```diff
@@ -1,5 +1,89 @@
+    param ( [Parameter(Mandatory=$true,Position=0)]$content )
+    $method = "PUT"
+    $headerDate = '2014-02-14'
+    $headerDate = '2015-12-11'
```
$headerDate overwritten?

JamesWTruher replied:
fixed

adityapatwardhan commented on tools/travis.ps1:
```diff
@@ -1,5 +1,89 @@
+        "`n" +
+        "`n" +
+        "`n" +
+        "`n"
```
Use a stringbuilder instead.

adityapatwardhan commented on tools/travis.ps1:
```diff
@@ -1,5 +1,89 @@
+    $signatureString += "x-ms-date:" + $headers["x-ms-date"] + "`n"
+    $signatureString += "x-ms-version:" + $headers["x-ms-version"] + "`n"
+    #Add CanonicalizedResource
+    $uri = New-Object System.Uri -ArgumentList $url
```
use System.Uri::new($url) instead

JamesWTruher replied:
fixed

adityapatwardhan commented on tools/travis.ps1:
```diff
@@ -1,5 +1,89 @@
+
+    $dataToMac = [System.Text.Encoding]::UTF8.GetBytes($signatureString)
+    $accountKeyBytes = [System.Convert]::FromBase64String($storageAccountKey)
+    $hmac = new-object System.Security.Cryptography.HMACSHA256((,$accountKeyBytes))
```
::new() here as well

JamesWTruher replied:
fixed

adityapatwardhan commented on tools/travis.ps1:
```diff
@@ -1,5 +1,89 @@
+
+    $headers["Authorization"]  = "SharedKey " + $storageAccountName + ":" + $signature
+
+    write-verbose $signatureString
```
Do you want to add [CmdletBinding()] to the function?

JamesWTruher replied:
fixed

JamesWTruher committed:
change new-object calls to use constructor

Remove extraneous date setting
update string creation to use stringbuilder rather than string addition
